src/main/kotlin/no/nav/faktureringskomponenten/service/EksternFakturaStatusService.kt

```diff
@@ -0,0 +1,72 @@
+package no.nav.faktureringskomponenten.service
+
+import mu.KotlinLogging
+import no.nav.faktureringskomponenten.domain.models.EksternFakturaStatus
+import no.nav.faktureringskomponenten.domain.models.Faktura
+import no.nav.faktureringskomponenten.domain.models.FakturaStatus
+import no.nav.faktureringskomponenten.domain.repositories.FakturaRepository
+import no.nav.faktureringskomponenten.exceptions.RessursIkkeFunnetException
+import no.nav.faktureringskomponenten.service.integration.kafka.ManglendeFakturabetalingProducer
+import no.nav.faktureringskomponenten.service.integration.kafka.dto.EksternFakturaStatusDto
+import no.nav.faktureringskomponenten.service.integration.kafka.dto.ManglendeFakturabetalingDto
+import no.nav.faktureringskomponenten.service.mappers.EksternFakturaStatusMapper
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+private val log = KotlinLogging.logger { }
+
+@Service
+class EksternFakturaStatusService(
+    private val fakturaRepository: FakturaRepository,
+    private val eksternFakturaStatusMapper: EksternFakturaStatusMapper,
+    private val manglendeFakturabetalingProducer: ManglendeFakturabetalingProducer
+) {
+
+    @Transactional
+    fun lagreEksternFakturaStatusMelding(eksternFakturaStatusDto: EksternFakturaStatusDto) {
+        log.info("Mottatt $eksternFakturaStatusDto")
+
+        val faktura = fakturaRepository.findById(eksternFakturaStatusDto.fakturaReferanseNr.toLong())
```

Comment:
Vi bør ikke bruke våre database ID-er som referanser utenfor.
Hvis du endrer / migrerer databasen kan du havne i problemer.
Database-ID har en funksjon og ref. for OEBS har en annen funksjon.
Det er best å ikke koble disse og lage egne ID-er i stedet.
Må ikke fikses i denne PR-en men må fikses før prodsetting, synes jeg

Reply:
Jeg lager en ny jira-bug som handler om refaktorering ang. dette. Det er nok flere steder dette oppstår, det er best å ta akkurat denne biten i en annen pr så denne ikke blir altfor stor.

https://jira.adeo.no/browse/MELOSYS-6186

Reply:
Takk. Enig at det blir bedre i egen PR.